text/0028-csv-indexation.md

```diff
@@ -0,0 +1,156 @@
+
+Since most of SQL engines or SQL clients can easily dump data as a csv file, it will facilitate MeiliSearch adoption by extending the indexation step on a wider range of customer cases than before.
+
+Writing performance is also considered as a motivation since CSV parsing is less CPU and memory intensive than parsing Json due to the streamable capability of the CSV format.
+
+### III. Additional Materials
+N/A
+
+### IV.Explanation
+
+#### Csv Formatting Rules
+
+While there's [RFC 4180](https://tools.ietf.org/html/rfc4180) as a try to add a specification for csv format, we will find a lot of variations from that. MeiliSearch features capabilities requires the csv file to be formatted the proper way to be parsable by the engine.
+
+- A csv file need to contain a first line representing the list of attributes with the choosen type separated from the attribute name by `:` character. 
+
+> An attribute can be specificed with two types: `string` or `number`. A missing type will be interpreted as a `string` by default.
+>
+> Valid headline example: "id:number","title:string","author","price:number"
+
+- Each of the following lines will represent a document for MeiliSearch.
+- Each value should be enclosed in double quotes tags to handle comma character presence inside.
+- Using double-quotes to enclose fields, then a double-quote appearing inside a field must be escaped by preceding it with another double quote as mentioned in [RFC 4180](https://tools.ietf.org/html/rfc4180).
+- Float value should be written with a `.` character, like `3.14`.
+- CSV text should be encoded in UTF8.
+- The format can't handle array cell values. We are providing `nd-json` format to deal with theses types of attribute in a easier way.
+
+##### Example with a comma inside a cell
+
+Given the csv payload
+```
+"id:number","label","price:number","colors","description"
+"1","t-shirt","4.99","red","Thus, you will rock at summer time."
+```
+the search result should be displayed as 
+```json
+{
+  "hits": [
+    {
+      "id": 1,
+      "label": "t-shirt",
+      "price": 4.99,
+      "colors": "red",
+      "description": "Hey, you will rock at summer time.",
+    }
+  ],
+  ...
+}
+```
+
+##### Example with a double quote inside a cell
+
+Given the csv payload
+```
+"id:number","label","price","colors","description"
+"1","t-shirt","4.99","red","Hey, you will ""rock"" at summer time."
+```
+the search result should be displayed as
+```json
+{
+  "hits": [
+    {
+      "id": 1,
+      "label": "t-shirt",
+      "price": "4.99",
+      "colors": "red",
+      "description": "Hey, you will rock at summer time.",
+    }
+  ],
+  ...
+}
+```
+
+> Note that the price attribute was not typed as a number. By default, MeiliSearch type it as a string.
+
+#### API Endpoints
+
+> Each API endpoints mentioned above will now require a `text/csv` as `Content-Type` header to process CSV data.
```

Comment:
Currently, the users don't need to use the header `Content-type: application/json` when adding documents in JSON because MeiliSearch infers they want to use JSON.
Is still `Content-type: application/json` the default header if no header is passed or should the users always pass a header with the type from now?

Reply:
We could keep this default behavior and make sure that if `Content-Type: application/json` is set, we have the exact same behavior.

Reply:
We should specify this behavior in the spec (sorry if it's already present, but I don't see it). Also, this should be present in the `impact on the Documentation` section. The documentation should describe this default behavior somewhere I think. It's up to tommy to decide, but I would rather he's aware of this behavior.

Reply:
I have added that part. Nice catch! Same for NDJSON specification.